app/start_handlers.py

```diff
@@ -238,6 +243,9 @@ class IndexEN(Start):
         session['case'] = uac_json
         session['attributes']['display_region'] = 'en'
 
+        if data.get('adlocation'):
```

Comment:
I can't see a scenario where the adlocation would remove the adlocation from the URL because they deliberately didn't want it?
The adlocation value is not 'client' specific but location specific, so would not be an issue if it is carried over from a previous client, although if the user has had either the session auto timeout, or used save-and-exit from EQ resume later the session is invalidated anyway.

Reply:
As discussed, seems to be desired behaviour the _adlocation_ attribute is not removed from a current session even if the /start/ URL is submitted without the relevant query parameter. So all seems good.